cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -132,6 +126,18 @@ public void setAuditPublisher(AuditPublisher auditPublisher) {
    * @throws IOException if there is a problem in making an HTTP request to check if the namespace exists.
    */
   public DatasetMeta get(Id.DatasetInstance instance, List<? extends Id> owners) throws Exception {
+    try {
+      return metaCache.get(instance);
+    } catch (ExecutionException e) {
+      Throwables.propagateIfPossible(e.getCause(), Exception.class);
```

Comment:
However, if the Exception is a NotFoundException, then the caller will see an ExecutionException. ExecutionException is only used by the cache to wrap the actual exception from the cache loader.